tasks/apache.yml

```diff
@@ -1,31 +1,59 @@
 ---
+# backuppc_server tasks to configure Apache2 for BackupPC access
 
-- name: Add module
+- name: APACHE2_MODULE | Add module
   apache2_module:
+    name: "{{ module }}"
     state: present
-    name: "{{item}}"
-  with_items: 
-    - rewrite
-    - ssl
+  loop:
+  - rewrite
+  - ssl
+  loop_control:
+    loop_var: module
     
-- name: Copy apache config
+- name: TEMPLATE | Copy apache config
   template:
-    dest: "/etc/apache2/sites-available/{{ backuppc_server_name }}.conf"
+    dest: "/etc/apache2/sites-available/{{ backuppc_srv_server_name }}.conf"
     src: etc/apache/apache.conf.j2
     owner: root
     group: root
     mode: "0644"
-  notify: restart apache2
+  notify: BackupPC_server - restart apache2
 
-- name: Remove default apache2 site.
+- name: FILE | Remove default apache2 site.
   file:
     path: /etc/apache2/sites-enabled/000-default.conf
     state: absent
-  notify: restart apache2
+  notify: BackupPC_server - restart apache2
+  when: backuppc_srv_disable_apache_default_site | bool
 
-- name: Activate custom Backuppc apache site.
+- name: FILE | Activate custom Backuppc apache site.
   file:
-    path: "/etc/apache2/sites-enabled/{{ backuppc_server_name }}.conf"
-    src: "/etc/apache2/sites-available/{{ backuppc_server_name }}.conf"
+    path: "/etc/apache2/sites-enabled/{{ backuppc_srv_server_name }}.conf"
+    src: "/etc/apache2/sites-available/{{ backuppc_srv_server_name }}.conf"
     state: link
-  notify: restart apache2
\ No newline at end of file
+  notify: BackupPC_server - restart apache2
+
+- name: LINEINFILE | Allow distant access to Apache2 BackupPC interface
+  lineinfile:
+    path: '{{ backuppc_srv_config_dir }}/apache.conf'
+    regex: '#? ?Require local'
+    line: '                # Require local'
+    state: present
+  notify: BackupPC_server - restart apache2
+  when: 
+  - not backuppc_srv_apache_local | bool
+  - ansible_facts['distribution'] == "Debian"
+  - ansible_facts['distribution_major_version'] | int >= 10
+
+
+- name: LINEINFILE | Require SSL to access to Apache2 BackupPC interface
+  lineinfile:
+    path: '{{ backuppc_srv_config_dir }}/apache.conf'
+    regex: '#?SSLRequireSSL'
+    line: '        SSLRequireSSL'
+    state: present
+  notify: BackupPC_server - restart apache2
+  when: backuppc_srv_apache_require_ssl | bool
```

Comment:
These two tasks don't work well. There is some problem with regex, because they let `/etc/backuppc/apache.conf` with the following content: 
```Apache
Alias /backuppc /usr/share/backuppc/cgi-bin/

<Directory /usr/share/backuppc/cgi-bin/>
        AllowOverride None

        # Uncomment the line below to ensure that nobody can sniff important
        # info from network traffic during editing of the BackupPC config or
        # when browsing/restoring backups.
        # Requires that you have your webserver set up for SSL (https) access.
        #SSLRequireSSL
                                        
        Options ExecCGI FollowSymlinks
        AddHandler cgi-script .cgi
        DirectoryIndex index.cgi

        AuthUserFile /etc/backuppc/htpasswd
        AuthType basic
        AuthName "BackupPC admin"

        <RequireAll>
        SSLRequireSSL
                Require local

                # This line ensures that only authenticated users may access your backups
                Require local
        </RequireAll>
</Directory>
```

Reply:
My regex are correct as grep extended regex, not basic ones: 
```
root@anacahuita:~# grep -E '#? ?Require ' /etc/backuppc/apache.conf 
		Require local
                Require local
root@anacahuita:~# grep '#? ?Require ' /etc/backuppc/apache.conf 
root@anacahuita:~# grep -E '#?SSLRequireSSL' /etc/backuppc/apache.conf 
        #SSLRequireSSL
        SSLRequireSSL
root@anacahuita:~# grep '#?SSLRequireSSL' /etc/backuppc/apache.conf 
```
[lineinfile ansible module](https://docs.ansible.com/ansible/latest/modules/lineinfile_module.html#lineinfile-module) uses [Python regular expressions](https://docs.python.org/2/library/re.html). What's wrong with mine in this RE dialect??

Reply:
Probando las regex de las tareas con el contenido del archivo en [pythex.org/](https://pythex.org/), no hay problema con esas regex... ¿¿Qué le pasa a Ansible??

Reply:
Regex where working ok, but my first regex, in all cases, also met: 
```
# Comment out this line once you have setup HTTPS and uncommented SSLRequireSSL
```